crates/bindings/src/table.rs

```diff
@@ -388,6 +388,30 @@ impl<Tbl: Table, Col: Index + Column<Table = Tbl>> UniqueColumn<Tbl, Col::ColTyp
         let buf = IterBuf::take();
         update::<Tbl>(Col::index_id(), new_row, buf)
     }
+
+    /// Attempts to insert `new_row` into the table. If it exists, deletes the row with the same value in the unique column.
+    #[track_caller]
+    #[cfg(feature = "unstable")]
+    pub fn try_upsert(&self, new_row: Tbl::Row) -> Result<Tbl::Row, TryInsertError<Tbl>> {
```

Comment:
"Upsert" is the colloquial name for this operation, but I'd prefer the more explicit `try_insert_or_update`, and `insert_or_update` for the panicking version.

If you like, you could also add a [rustdoc alias](https://doc.rust-lang.org/rustdoc/advanced-features.html#add-aliases-for-an-item-in-documentation-search) with the name `try_upsert` (and `upsert` on the panicking version) to aid discovery.